sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/slack.yaml

```diff
@@ -0,0 +1,92 @@
+provider:
+  metadata:
+    name: Slack Webhook URL
+    category: Messaging System
+    message: Make sure this Slack Webhook URL gets revoked, changed, and removed from the code.
+
+  rules:
+    - rspecKey: S6713
+      id: slack-incoming-webhook-urls
+      metadata:
+        name: Slack Incoming Webhook URL
+      examples:
+        - text: |
+            // Use the url you got earlier
+            $slack = new Slack('https://hooks.slack.com/services/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');
+
+            // Create a new message
+            $message = new SlackMessage($slack);
+            $message->setText("Hello world!");
+
+            // Send it!
+            if ($message->send()) {
+                echo "Hurray 😄";
+            } else {
+                echo "Failed 😢";
+            }
+          containsSecret: true
+          match: https://hooks.slack.com/services/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
+        - text: |
+            // Use the url you got earlier
+            $slack = new Slack('{slack_webhook_url}');
+
+            // Create a new message
+            $message = new SlackMessage($slack);
+            $message->setText("Hello world!")->setChannel("#general");
+
+            // Send it!
+            $message->send();
+          containsSecret: false
+      detection:
+        matching:
+          # the structure is:
+          # https://hooks.slack.com/(services/)?{TEAM_ID}/{WEBHOOK_ID}/{WEBHOOK_TOKEN}
+          pattern: "((https://)?hooks\\.slack\\.com/services/T[A-Za-z0-9+\\/]{42,45})"
+        post:
+          # Avoid matching values found on SourceGraph that look like dummy passwords or insertions like:
+          # - https://hooks.slack.com/services/T00000000/B00000000/XXXXXXXXXXXXXXXXXXXXXXXX
+          patternNot: 
+            "[0]{6,}|\
+            [X]{6,}"
```

Comment:
You can group this one and the other one in the provider block.

Also, I think we should have the habit of having test cases that should fail because of these. I used this to test, and looked like it worked.

```yaml
        - text: |
            // Use the url you got earlier
            $slack = new Slack('https://hooks.slack.com/services/xxxxxxxxx/xxxxxxxxx/xxxxxxxxxxxxxxxxxxxxxxxx');
            // Create a new message
            $message = new SlackMessage($slack);
            $message->setText("Hello world!")->setChannel("#general");
             // Send it!
             $message->send();
           containsSecret: false
```